cobbler/settings/migrations/V3_0_1.py

```diff
@@ -38,14 +38,43 @@ def normalize(settings: dict) -> dict:
     return schema.validate(settings)
 
 
+def __migrate_modules_conf():
+    modules_conf_path = "/etc/cobbler/modules.conf"
+    with open(modules_conf_path, "r") as modules_conf_file:
+        modules_conf = modules_conf_file.readlines()
+
+    modules_conf_result = []
+    for line in modules_conf:
+        regex_authentication = re.compile("authn_")
```

Comment:
why are there regular expressions? as far as I can tell we're only looking for fixed substrings at the beginning of a line.

Reply:
If you look at the `modules.conf` file then you will see that we will also have usages in comments as well as usages that are not at the start of a line. So if I am not mistaken we need them as regexes?

Source: https://github.com/cobbler/cobbler/blob/b6617e44c4b16ea3efd6ea05e8b0b51983edfc82/config/cobbler/modules.conf#L23

Reply:
Either we do a substring search ([`str.find`](https://docs.python.org/3/library/stdtypes.html#str.find) since we need the position to replace), or we use regular expression but don't just match the beginning of strings.

Reply:
Okay I will reimplement this part of the script with `str.find`. Since the usage of this can be both in comments and the configuration itself a regex would be too complex. As we know that the file is small using `str.find` is the better way in my eyes.

Reply:
Okay refactoring is done. If you have time, please have a look.